src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -261,10 +290,12 @@ void GcsNodeManager::OnNodeFailure(const NodeID &node_id) {
     node_info_delta->set_state(node->state());
     node_info_delta->set_end_time_ms(node->end_time_ms());
 
-    auto on_done = [this, node_id, node_info_delta](const Status &status) {
-      RAY_CHECK_OK(gcs_publisher_->PublishNodeInfo(node_id, *node_info_delta, nullptr));
+    auto on_done = [this, node_id, callback, node_info_delta](const Status &status) {
+      RAY_CHECK_OK(gcs_publisher_->PublishNodeInfo(node_id, *node_info_delta, callback));
```

Comment:
We don't need to do the callback after `PublishNodeInfo` is done. I think there is no ordering requirement here. We can just invoke the callback directly in `on_done`.